src/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { initializeApp, FirebaseApp, FirebaseOptions } from "firebase/app";
+import * as ReactDOM from "react-dom";
+import { Auth, getAuth } from "firebase/auth";
+import {
+  enableIndexedDbPersistence,
+  Firestore,
+  getFirestore,
+} from "firebase/firestore";
+
+import { App } from "./components/App";
+import * as serviceWorker from "./serviceWorker";
+
+const devConfig: FirebaseOptions = {
+  projectId: "mapeo-webmaps-staging",
+  appId: "1:354071501370:web:a92ea6497d55c4dd9ab303",
+  storageBucket: "mapeo-webmaps-staging.appspot.com",
+  apiKey: "AIzaSyAddUwtJxCTq3VImtID0S-5beOHiJtKTe4",
+  authDomain: "mapeo-webmaps-staging.firebaseapp.com",
+  messagingSenderId: "354071501370",
+  measurementId: "G-GLMDFRWPJD",
+};
+
+const getConfig = () =>
+  fetch("/__/firebase/init.json").then(async (response) =>
+    response.json()
+  ) as FirebaseOptions;
+
+export let db: Firestore;
+export let auth: Auth;
+
+export let firebaseApp: FirebaseApp;
```

Comment:
Does this work? Not entirely sure you can have mutable exports in ES6 modules, but since Typescript is compiling to commonJS you might be able to get away with it.

The alternative is:

```ts
let db: Firestore
let auth: Auth
let firebaseApp: FirebaseApp

export const firebase = () => {
  if (!auth || !firebaseApp || !db) {
    throw new Error("App is not yet initialized")
  }
  return { db, auth, app: firebaseApp }
}
```

Reply:
(Hi sorry I thought I'd replied already!)

So this works currently like this, like you say this is likely because it's compiling down to commonjs. 
Ultimately it works the same as it was working before, ie. blocking rendering the whole react app until firebase is ready. Its fine but will make some UX improvements (probably a skeleton ui state) further down the line.

Reply:
(hi sorry I thought I'd already replied!)

So this works in it's current implementation, as you said thats likely because it's compiling to commonjs.

Reply:
related top this – the current implementation ultimately works like the old version - ie. the react app is fully blocked from rendering until firebase is ready. Whilst this is only a split second in practice, it's probably worth refactoring eventually. 

Further down the line (not this PR) I'll create some kind of skeleton UI that can be displayed as a loading screen with whilst firebase is initiated. But as I said that's for a later PR, I have added an issue #72 for it for it